### DISCUSSION TITLE
How do I use this with FastAPI?

### DISCUSSION THREAD

**Discussion by tyteen4a03** [nextauthjs/next-auth#2503] [Help]
Hi, I'm trying to create a React SPA that talks to a FastAPI and logs in using our existing OAuth2 server (custom, on Laravel Passport). It seems like NextAuth will kill a lot of my OAuth2 headaches so I'm keen on using it. 

Is it possible to only leverage the frontend of NextAuth and reimplement whatever NextAuth needs to store the sessions in FastAPI?

**Comment by balazsorban44** [comment]
NextAuth.js has its strength in its backend/server-side implementation, and actually it's the frontend that is kind of just a tiny wrapper for it. I looked it  up, and looks like FastAPI is written in Python, which we don't have plans for supporting at the moment.
